components/core/src/clp/string_utils/string_utils.hpp

```diff
@@ -61,6 +61,16 @@ std::string replace_characters(
  */
 void to_lower(std::string& str);
 
+/**
+ * Replace unescaped instances of src_char with target_char in src_string
+ * @param escape_char The character which indicates escape
+ * @param src_char The character which should be replaced if unescaped
+ * @param target_char The character which should replace src_char
+ * @param search_string The string which needs to have src_char replaced
+ */
+void
+replace_unescaped_char(char escape_char, char src_char, char target_char, std::string& src_string);
```

Comment:
Another nit. let's use trailinh return.
```suggestion
auto
replace_unescaped_char(char escape_char, char src_char, char target_char, std::string& src_string) -> void;
```

Reply:
I believe the trailing return type has more value when it comes to some complicated templated functions, and it works better with auto. For this function though, using trailing doesn't have any functional impact. It's mainly to align with our new coding standard. The string_utils might be written at the time before we use this standard so its not update-to-date.